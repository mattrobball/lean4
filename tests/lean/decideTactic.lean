/-!
# Tests of the `decide` tactic
-/

/-!
Success
-/
example : 2 + 2 ≠ 5 := by decide

/-!
False proposition
-/
example : 1 ≠ 1 := by decide

/-!
Irreducible decidable instance
-/
opaque unknownProp : Prop

open scoped Classical in
example : unknownProp := by decide
