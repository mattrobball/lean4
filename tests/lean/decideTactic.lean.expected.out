decideTactic.lean:13:22-13:28: error: tactic 'decide' proved that the proposition
  1 ≠ 1
is false
decideTactic.lean:21:28-21:34: error: tactic 'decide' failed for proposition
  unknownProp
since its 'Decidable' instance reduced to
  Classical.choice ⋯
rather than to the 'isTrue' constructor.
